dali/operators/util/reshape.cc

```diff
@@ -28,34 +29,71 @@ DALI_SCHEMA(Reshape)
   .AllowSequences()
   .SupportVolumetric()
   .AddOptionalArg<int>("shape", "The desired shape of the output. Number of elements in "
-                                "each sample must match that of the input sample.",
+                                "each sample must match that of the input sample. There can be "
+                                "one negative extent which receives the size required to match "
+                                "the input volume.",
                                 std::vector<int>(), true)
-  .AddOptionalArg("layout",     "New layout for the data. If not specified, the output layout "
-                                "is preserved if number of dimension matches existing layout "
-                                "or reset to empty otherwise",
-                                "");
+  .AddOptionalArg<float>("rel_shape", "The relative shape of the output. Number of dimensions "
+                  "must not exceed the number of dimensions of the input. There can be one "
+                  "negative value which means all remaining extents, e.g. input of shape "
+                  "`[480, 640, 3]` and `rel_shape = [0.5, -1]` would get the shape [240, 3840].",
+                                std::vector<float>(), true)
+  .AddOptionalArg("layout", "New layout for the data. If not specified, the output layout "
+                            "is preserved if number of dimension matches existing layout "
+                            "or reset to empty otherwise",
+                            "");
 
 
 template <typename Backend>
 Reshape<Backend>::Reshape(const OpSpec &spec) : Base(spec) {
   bool has_shape_input = spec.NumRegularInput() == 2;
-  bool has_shape_arg = spec.HasArgument("shape");
+  bool has_shape_arg = spec.HasArgument("shape") || spec.HasTensorArgument("shape");
   bool has_layout_arg = spec.HasArgument("layout");
-  DALI_ENFORCE(!(has_shape_input && has_shape_arg),
-    "Reshape: use either shape input or shape argument, not both");
-  DALI_ENFORCE(has_shape_input || has_shape_arg || has_layout_arg,
+  bool has_rel_shape_arg = spec.HasArgument("rel_shape") || spec.HasTensorArgument("rel_shape");
+  DALI_ENFORCE(has_shape_input + has_shape_arg + has_rel_shape_arg == 1,
+    "Reshape: shape input, `shape` argument and `rel_shape` argument are mutually exclusive");
+  DALI_ENFORCE(has_shape_input || has_shape_arg || has_rel_shape_arg || has_layout_arg,
```

Comment:
Note: you are already checking that at least one shape argument is provided, you don't need to check that again.

Reply:
There's a bug - it should be <= 1.